presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1570,6 +1571,7 @@ private Scope computeAndAssignOrderByScope(OrderBy node, Scope sourceScope, Scop
             // ORDER BY should "see" both output and FROM fields during initial analysis and non-aggregation query planning
             Scope orderByScope = Scope.builder()
                     .withParent(sourceScope)
+                    .withScopeId(outputScope.getScopeId())
```

Comment:
You could maybe have some `QueryRelation` or `NamedRelation` what would compose of both `RelationId` and `RelationType`. This could simplify things a bit since those two things often come in pairs.